internal/engine/compiler/engine.go

```diff
@@ -1065,12 +1076,29 @@ entry:
 			stack := ce.stack[base : base+stackLen]
 
 			fn := calleeHostFunction.parent.goFunc
-			switch fn := fn.(type) {
-			case api.GoModuleFunction:
-				fn.Call(ctx, ce.callerModuleInstance, stack)
-			case api.GoFunction:
-				fn.Call(ctx, stack)
-			}
+			func() {
+				if snapshotEnabled {
+					defer func() {
+						if r := recover(); r != nil {
+							if s, ok := r.(*snapshot); ok {
+								if s.ce == ce {
```

Comment:
Can we do a single `ok && s.ce == ce` and merge the else clauses?

Reply:
Sweet, thanks!